tds-itemselection-impl/src/main/java/tds/itemselection/msb/MsbAssessmentSelectionServiceImpl.java

```diff
@@ -0,0 +1,145 @@
+/*******************************************************************************
+ * Educational Online Test Delivery System
+ * Copyright (c) 2016 Regents of the University of California
+ *
+ * Distributed under the AIR Open Source License, Version 1.0
+ * See accompanying file AIR-License-1_0.txt or at
+ * http://www.smarterapp.org/documents/American_Institutes_for_Research_Open_Source_Software_License.pdf
+ *
+ * SmarterApp Open Source Assessment Software Project: http://smarterapp.org
+ * Developed by Fairway Technologies, Inc. (http://fairwaytech.com)
+ * for the Smarter Balanced Assessment Consortium (http://smarterbalanced.org)
+ ******************************************************************************/
+
+package tds.itemselection.msb;
+
+import AIR.Common.DB.SQLConnection;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Service;
+import tds.itemselection.api.IItemSelection;
+import tds.itemselection.base.ItemCandidatesData;
+import tds.itemselection.base.ItemGroup;
+import tds.itemselection.base.TestItem;
+import tds.itemselection.loader.IItemSelectionDBLoader;
+import tds.itemselection.loader.SegmentCollection2;
+import tds.itemselection.loader.TestSegment;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+@Service
+public class MsbAssessmentSelectionServiceImpl implements MsbAssessmentSelectionService {
+
+    public MsbAssessmentSelectionServiceImpl() {
+    }
+
+    public MsbAssessmentSelectionServiceImpl(IItemSelectionDBLoader itemSelectionDbLoader) {
+        this.itemSelectionDbLoader = itemSelectionDbLoader;
+    }
+
+    private final String FIXED_ALGORITHM = "fixedform";
+
+    @Autowired
```

Comment:
Can we @Autowired the constructor passing in everything rather than relying on field autowiring.  We could then also remove empty constructor.